Admin/AdminHelper.php

```diff
@@ -113,42 +116,67 @@ public function appendFormFieldElement(AdminInterface $admin, $subject, $element
         // get the field element
         $childFormBuilder = $this->getChildFormBuilder($formBuilder, $elementId);
 
-        // retrieve the FieldDescription
-        $fieldDescription = $admin->getFormFieldDescription($childFormBuilder->getName());
+        //Child form not found (probably nested one)
+        //if childFormBuilder was not found resulted in fatal error getName() method call on non object
+        if (!$childFormBuilder) {
+            $propertyAccessor = new PropertyAccessor();
+            $entity = $admin->getSubject();
 
-        try {
-            $value = $fieldDescription->getValue($form->getData());
-        } catch (NoValueException $e) {
-            $value = null;
-        }
+            $path = $this->getElementAccessPath($elementId, $entity);
 
-        // retrieve the posted data
-        $data = $admin->getRequest()->get($formBuilder->getName());
+            $collection = $propertyAccessor->getValue($entity, $path);
 
-        if (!isset($data[$childFormBuilder->getName()])) {
-            $data[$childFormBuilder->getName()] = array();
-        }
+            if ($collection instanceof ArrayCollection) {
+                $entityClassName = $this->getEntityClassName($admin, explode('.', preg_replace('#\[\d*?\]#', '', $path)));
+            } elseif ($collection instanceof \Doctrine\ORM\PersistentCollection) {
```

Comment:
missing the fix for mongodb
https://github.com/sonata-project/SonataAdminBundle/pull/2985#discussion-diff-31056069

```php

        if ($collection instanceof \Doctrine\ORM\PersistentCollection || $collection instanceof \Doctrine\ODM\MongoDB\PersistentCollection) {
            //since doctrine 2.4
            $entityClassName = $collection->getTypeClass()->getName();
        } elseif ($collection instanceof \Doctrine\Common\Collections\Collection) {
            $entityClassName = $this->entityClassNameFinder($admin, explode('.', preg_replace('#\[\d*?\]#', '', $path)));
        } else {
            return;
        }
```

with this change it should work for all possible mappers of doctrine (including the easy check for ORM and ODM)